optimum/intel/utils/modeling_utils.py

```diff
@@ -229,4 +238,209 @@ def bind_cores_for_best_perf():
+        else:
+            model = open_clip.CustomTextCLIP(**model_cfg, cast_dtype=open_clip.get_cast_dtype(dtype))
+
+        open_clip.load_checkpoint(model, model_path)
+
+        if "preprocess_cfg" in open_clip_config:
+            preprocess_cfg = open_clip.get_model_preprocess_cfg(model)
+            for k, val in open_clip_config["preprocess_cfg"].items():
+                preprocess_cfg[k] = val
+            open_clip.set_model_preprocess_cfg(model, preprocess_cfg)
+
+        return model
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        model_name_or_path: Union[str, Path],
+        config: Optional["PretrainedConfig"] = None,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        trust_remote_code: bool = False,
+        dtype: str = "f32",
+        **kwargs,
+    ):
+        if not is_open_clip_available():
+            raise ImportError(
+                "To load a open_clip model, open_clip needs to be installed. Please install it with `pip install open-clip-torch`."
+            )
+
+        import open_clip
+
+        if os.path.isdir(model_name_or_path):
+            local_open_clip_model = collect_open_clip_model_files(model_name_or_path)
+            if "model" in local_open_clip_model and "open_clip_config" in local_open_clip_model:
+                model_config, config_as_dict = cls.load_config_from_file(
+                    local_open_clip_model["open_clip_config"], model_name_or_path
+                )
+
+                model = cls.create_open_clip_model(local_open_clip_model["model"], config_as_dict, dtype)
+                setattr(model, "config", model_config)
+
+                if not getattr(model.config, "model_type", None) and not getattr(
+                    model.config, "export_model_type", None
+                ):
+                    setattr(model.config, "model_type", "clip")
+                    setattr(model.config, "export_model_type", "clip")
+
+                return model
+            else:
+                raise IOError(
+                    f"Fail to load open_clip model from path {model_name_or_path}. Folder should contains open_clip_config.json and open_clip_pytorch_model.bin"
+                )
+        else:
+            try:
+                model, _ = open_clip.create_model_from_pretrained(
+                    f"hf-hub:{model_name_or_path}", cache_dir=cache_dir, force_custom_text=True
+                )
+
+                if not getattr(model, "config", None):
+                    config_path = hf_hub_download(
+                        repo_id=model_name_or_path, filename="open_clip_config.json", cache_dir=cache_dir
```

Comment:
we could also give argument like revision, subfolder, token, force_download, trust_remote_code, local_files_only